src/main/kotlin/org/rust/ide/intentions/CreateFunctionIntention.kt

```diff
@@ -37,16 +38,25 @@ class CreateFunctionIntention : RsElementBaseIntentionAction<CreateFunctionInten
 
     sealed class Context(val name: String, val callElement: PsiElement) {
         abstract val visibility: String
+        open val isAsync: Boolean = callElement.parentsOfType<RsBlock>().firstOrNull().let { block ->
```

Comment:
Do we really want to check if containing function is `async`? Code like this:
```rust
fn foo() {
    bar().await;
}
```
is invalid, but if user already has `await`, maybe user also will make function `foo` be `async` later?

Reply:
FWIW I was thinking the same, but I didn't feel strongly either way. If there is no check for async, there should probably at least be a check for edition 2018.

Reply:
Well, let me modify it to only detect edition 2018.